### PR TITLE
Add GPG + pinentry

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

Closes https://github.com/nebari-dev/nebari/issues/2000

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: .github/workflows/build-push-docker.yaml [hunk 1]
```diff
@@ -97,12 +98,7 @@ jobs:
             "quay.io/${{ env.DOCKER_ORG }}/nebari-${{ matrix.dockerfile }}${{ env.IMAGE_SUFFIX }}"
             "ghcr.io/${{ github.repository_owner }}/nebari-${{ matrix.dockerfile }}${{ env.IMAGE_SUFFIX }}"
           tags: |
-            # branch event -> e.g. `main-f0f6994-20221001`
-            type=ref, event=branch, suffix=-{{sha}}-{{date 'YYYYMMDD'}}
-            # needed for integration tests
-            type=ref, event=branch
-            # on tag push -> e.g. `2022.10.1`
-            type=ref, event=tag
+            20230914
```
Comment (pavithraes) [inline .github/workflows/build-push-docker.yaml:101]: Could you please share why are we hardcoding the tag? (I'm not super familiar with docker, so apologies if this is obvious!)

File: .github/workflows/build-push-docker.yaml [hunk 2]
```diff
@@ -115,7 +111,7 @@ jobs:
           context: .
           file: "Dockerfile.${{ matrix.dockerfile }}"
           tags: ${{ steps.meta.outputs.tags }}
-          push: ${{ github.event_name != 'pull_request' }}
+          push: true
```
Comment (pavithraes) [inline .github/workflows/build-push-docker.yaml:114]: This means we'll be building docker images on PRs as well, correct? To confirm, is this required/intentional?

Comment (iameskild) [reply .github/workflows/build-push-docker.yaml:114]: This is only a temporary change (alongside this temporary commit) so we can test the changes. I will revert this change after we've confirmed that the new image works as expected.

I just [opened this](https://github.com/nebari-dev/nebari-docker-images/issues/87) issue to discuss whether we want to push images to Quay on a PR trigger. Thanks for asking, this is a good point for us to discuss in more detail as a team.

File: .github/workflows/build-push-docker.yaml (hunk 1 above)
Comment (iameskild) [reply .github/workflows/build-push-docker.yaml]: See note below. Change has been reverted.

File: .github/workflows/build-push-docker.yaml (hunk 2 above)
Comment (iameskild) [reply .github/workflows/build-push-docker.yaml]: This change has now been reverted since the images are now available on Quay for us to test with :)

**Comment by iameskild** [comment]
@kcpevey @pavithraes I was able to successfully [create a GPG key](https://docs.gitlab.com/ee/user/project/repository/signed_commits/gpg.html#create-a-gpg-key) while using this updated jupyterlab image.

**Review by kcpevey — APPROVED** [review]

**Pull request merged**